src/python/utils/diff_statistics.py

```diff
@@ -0,0 +1,167 @@
+from pathlib import Path
+from typing import Union, Set
+
+from loguru import logger
+
+from src.python.repo import Repository
+
+
+def get_diff(
+    repo: Repository, left_sha: str, right_sha: str, diff_log_file: Union[None, Path]
+) -> str:
+    """
+    Computes the diff between two branches using git diff.
+    Args:
+        repo (Repository): The repository object.
+        left_sha (str): The left sha.
+        right_sha (str): The right sha.
+    Returns:
+        str: A string containing the diff result.
+    """
+    command = f"git diff {left_sha} {right_sha}"
+    stdout, _ = repo.run_command(command)
+    return stdout
+
+
+def get_diff_files_branches(repo: Repository, left_sha: str, right_sha: str) -> set:
```

Comment:
What is the distinction between `get_diff_files_branches` and `get_diff_files_merge`?  Improved documentation would help.  Also, consider changing "diff" to "differing" unless it really refers to a git diff specifically.